src/mongodb/scores.controller.ts

```diff
@@ -699,137 +1572,1444 @@ export class ScoresController {
+        if (totalValidation > 2) {
+          currentLevel = 2;
+        } else {
+          currentLevel = 3;
+        }
+      } else if (totalTargets <= 2) {
+        if (totalValidation > 0) {
+          currentLevel = 3;
+        } else {
+          currentLevel = 4;
+        }
+      }
+    }
+
+    let sessionHighestLevel: any = 0;
+    let sessionHighestTargets = 0;
+    let sessionHighestValidations = 0;
+
+    for (let sessionEle of sessions) {
+
+      let sessionLevel = 0;
+      let sessionTargets: any = await this.scoresService.getTargetsBySession(sessionEle);
+      let sessionValidations: any = await this.scoresService.getAssessmentRecords(sessionEle);
+
+      let totalSessionTargets = sessionTargets.length;
+      let totalSessionValidations = sessionValidations.length;
+
+      if (totalSession === 0) {
+        sessionLevel = 0;
+      } else {
+        if (totalSessionTargets >= 30) {
+          sessionLevel = 1;
+        } else if (totalSessionTargets < 30 && totalSessionTargets >= 16) {
+          console.log(totalSessionTargets);
+          console.log(totalSessionValidations);
+          if (totalSessionValidations > 5) {
+            sessionLevel = 1;
+          } else {
+            sessionLevel = 2;
+          }
+        } else if (totalSessionTargets < 16 && totalSessionTargets > 2) {
+          if (totalSessionValidations > 2) {
+            sessionLevel = 2;
+          } else {
+            sessionLevel = 3;
+          }
+        } else if (totalSessionTargets <= 2) {
+          if (totalSessionValidations > 0) {
+            sessionLevel = 3;
+          } else {
+            sessionLevel = 4;
+          }
+        }
+      }
+
+      if (sessionHighestLevel < sessionLevel) {
+        sessionHighestLevel = sessionLevel;
+        sessionHighestTargets = totalSessionTargets;
+        sessionHighestValidations = totalSessionValidations;
+      }
+    }
+
+    console.log(sessionHighestLevel);
+
+    if (currentLevel < sessionHighestLevel) {
+      currentLevel = sessionHighestLevel;
+      totalTargets = sessionHighestTargets;
+      totalValidation = sessionHighestValidations;
+    }
+
+    currentLevel = 'm' + currentLevel;
+
+    return { currentLevel: currentLevel, targetsCount: totalTargets, validationsCount: totalValidation, targets: targets, validations: validations };
+  }
+
+  @ApiExcludeEndpoint(true)
+  @Post('/getSetResult')
+  async getSetResult(@Res() response: FastifyReply, @Body() getSetResult: any) {
+
+    let targets = await this.scoresService.getTargetsBysubSession(getSetResult.sub_session_id, getSetResult.contentType);
+
+    let totalTargets = targets.length;
+    let sessionResult = 'No Result';
+
+    let recordData: any = await this.scoresService.getlatestmilestone(getSetResult.user_id);
+    let previous_level = recordData[0]?.milestone_level || undefined;
+
+    if (getSetResult.contentType === 'Char' || getSetResult.contentType === 'char') {
```

Comment:
why the or comparison is repeated? both conditions are same.  
Applies to the below too.